pmd-java/src/main/java/net/sourceforge/pmd/lang/java/ast/internal/PrettyPrintingUtil.java

```diff
@@ -170,19 +170,19 @@ public static String getNodeName(JavaNode node) {
         // constructors are differentiated by their parameters, while we only use method name for methods
         if (node instanceof ASTMethodDeclaration) {
             return ((ASTMethodDeclaration) node).getName();
-        } else if (node instanceof ASTMethodOrConstructorDeclaration) {
+        } else if (node instanceof ASTExecutableDeclaration) {
             // constructors are differentiated by their parameters, while we only use method name for methods
             return displaySignature((ASTConstructorDeclaration) node);
         } else if (node instanceof ASTFieldDeclaration) {
             return ((ASTFieldDeclaration) node).getVarIds().firstOrThrow().getName();
         } else if (node instanceof ASTResource) {
             return ((ASTResource) node).getStableName();
-        } else if (node instanceof ASTAnyTypeDeclaration) {
-            return ((ASTAnyTypeDeclaration) node).getSimpleName();
-        } else if (node instanceof ASTVariableDeclaratorId) {
-            return ((ASTVariableDeclaratorId) node).getName();
+        } else if (node instanceof ASTTypeDeclaration) {
+            return ((ASTTypeDeclaration) node).getSimpleName();
+        } else if (node instanceof ASTVariableId) {
+            return ((ASTVariableId) node).getName();
         } else {
-            return node.getImage(); // todo get rid of this
+            throw new IllegalArgumentException("Node " + node + " has no defined name");
```

Comment:
Just a detail, but since the node toString might be large and contain spaces I think it's nicer to put it at the end
```suggestion
            throw new IllegalArgumentException("Node has no defined name: " + node);
```